sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ParameterNamesInPartialMethod.cs

```diff
@@ -50,9 +50,14 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     var methodSyntax = (MethodDeclarationSyntax)c.Node;
                     var methodSymbol = c.SemanticModel.GetDeclaredSymbol(methodSyntax);
 
-                    if (methodSymbol?.PartialDefinitionPart != null)
+                    if (methodSymbol?.PartialImplementationPart != null)

```

Comment:
This was already wrong with Roslyn 1.0 but there was a bug making roslyn call this method both for the implementation and declaration part (should have been only declaration) that's why the bug pops only now.